core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -400,7 +398,9 @@ private[kafka] class Processor(val id: Int,
 
   private val newConnections = new ConcurrentLinkedQueue[SocketChannel]()
   private val inflightResponses = mutable.Map[String, RequestChannel.Response]()
-  private val metricTags = Map("networkProcessor" -> id.toString).asJava
+  val metricTags = Map(
+    "protocol" -> securityProtocol.name, "listener" -> listenerName.value, "networkProcessor" -> id.toString
+    ).asJava
 
   newGauge("IdlePercent",
```

Comment:
Yes it was intentional, as "io-wait-ratio" is still a per-network-thread metric and JConsole was grouping it with KafkaMetrics, so I didn't spot it coming from a different registry.
IMHO it would look too odd to leave that just tagged with the processor thread number. I'll clarify the KIP